libgalois/include/katana/Statistics.h

```diff
@@ -434,9 +434,30 @@ ReportStatAvg(
 //! @param id Identifier to prefix stat with in statistics output
 KATANA_EXPORT void reportRUsage(const std::string& id);
 
-//! Reports Galois system memory stats for all threads
+//! Reports system memory stats for all threads
 KATANA_EXPORT void reportPageAlloc(const char* category);
 
+class [[nodiscard]] ReportPageAllocGuard {
+public:
+  ReportPageAllocGuard() { reportPageAlloc("MeminfoPre"); }
+  ~ReportPageAllocGuard() { Report(); }
+  ReportPageAllocGuard(const ReportPageAllocGuard&) = delete;
+  ReportPageAllocGuard(ReportPageAllocGuard &&) = delete;
+  ReportPageAllocGuard& operator=(const ReportPageAllocGuard&) = delete;
+  ReportPageAllocGuard& operator=(ReportPageAllocGuard&&) = delete;
+
+  void Report() {
+    if (reported_) {
```

Comment:
Why have a guard like this? And if it's needed why not only call this in the destructor to avoid that need?

Reply:
in some places, just like `Timer`, the timing regions are in the middle of blocks. I didn't want to move around measured regions in this PR, so `Report` is a way to keep the same measurements without too much change to existing code.